dask/dataframe/tests/test_multi.py

```diff
@@ -891,25 +935,43 @@ def test_concat3():
     ddf2 = dd.from_pandas(pdf2, 3)
     ddf3 = dd.from_pandas(pdf3, 2)
 
-    result = dd.concat([ddf1, ddf2])
+    with warnings.catch_warnings(record=True) as w:
+        expected = pd.concat([pdf1, pdf2])
+
+    ctx = FutureWarning if w else None
+
+    with pytest.warns(ctx):
+        result = dd.concat([ddf1, ddf2])
+
     assert result.divisions == ddf1.divisions[:-1] + ddf2.divisions
     assert result.npartitions == ddf1.npartitions + ddf2.npartitions
-    assert_eq(result, pd.concat([pdf1, pdf2]))
+    assert_eq(result, expected)
+
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore", FutureWarning)
+        assert_eq(dd.concat([ddf1, ddf2], interleave_partitions=True),
+                  pd.concat([pdf1, pdf2]))
+
+    with warnings.catch_warnings(record=True) as w:
+        expected = pd.concat([pdf1, pdf2, pdf3])
 
-    assert_eq(dd.concat([ddf1, ddf2], interleave_partitions=True),
-              pd.concat([pdf1, pdf2]))
+    ctx = FutureWarning if w else None
 
-    result = dd.concat([ddf1, ddf2, ddf3])
+    with pytest.warns(ctx):
+        result = dd.concat([ddf1, ddf2, ddf3])
     assert result.divisions == (ddf1.divisions[:-1] + ddf2.divisions[:-1] +
                                 ddf3.divisions)
     assert result.npartitions == (ddf1.npartitions + ddf2.npartitions +
                                   ddf3.npartitions)
-    assert_eq(result, pd.concat([pdf1, pdf2, pdf3]))
+    assert_eq(result, expected)
 
-    assert_eq(dd.concat([ddf1, ddf2, ddf3], interleave_partitions=True),
-              pd.concat([pdf1, pdf2, pdf3]))
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore", FutureWarning)
+        assert_eq(dd.concat([ddf1, ddf2, ddf3], interleave_partitions=True),
+                  pd.concat([pdf1, pdf2, pdf3]))
 
 
+@pytest.mark.filterwarnings('ignore')
```

Comment:
This was getting tedious, and I'm hoping the warning behavior is thoroughly tested by now.